### DISCUSSION TITLE
Explicit call of abstract static interface member (Preview feature)

### DISCUSSION THREAD

**Discussion by giladfrid009** [dotnet/csharplang#5027] [Q&A] [answered]
The lasest VS 2022 and .NET 6 provides access to preview language features such as abstract static interface member and generic numeric types.

I can't figure out how to call an explicit implementation of abstract interface member.
Suppose I have the following code:

```csharp
double ToDouble<T>(T val) where T : INumber<T>
{
    return double.Create(val); // does not compile. double implements INumber<T>.Create explicitly, how do I call it?
}

TRes Convert<T, TRes>(T val) where T : INumber<T>, where TRes : INumber<T>
{
    return TRes.Create(val); // compiles just fine, can provide double as TRes, and will act the same as the previous function.
}
```
How can I implement ``ToDouble`` directly?

**Comment by giladfrid009** [comment, marked as answer]
Apparently, invoking explicit static abstract method directly is not possible, so must use the Convert func.

**Reply by tannergooding** [reply]
Right. As the blog post (https://devblogs.microsoft.com/dotnet/preview-features-in-net-6-generic-math/) calls out, the members are all implemented explicitly for .NET 6 while the feature is in preview. We will look at making many of these members implicit and therefore publicly available in a future version of .NET (likely .NET 7).

Until then, you can only access these APIs via some generic type constrained to the relevant interface.
